packages/framework/presence/src/test/batching.spec.ts

```diff
@@ -0,0 +1,868 @@
+				}); // SIGNAL #1 - not queued because it contains a value manager with a latency of 0,
+				// so the initial data will be sent immediately
+
+				const { count, immediateUpdate } = stateWorkspace.props;
+
+				clock.tick(10); // Time is now 1020
+				count.local = { num: 12 }; // will be queued; deadline is set to 1120
+
+				clock.tick(80); // Time is now 1100
+				count.local = { num: 34 }; // will be queued; deadline remains 1120
+
+				clock.tick(10); // Time is now 1110
+				immediateUpdate.local = { num: 56 };
+				// SIGNAL #2
+				// This should cause the queued signals to be merged with this immediately-sent
+				// signal with the value from the last signal (num=34).
+
+				// It's necessary to tick the timer beyond the deadline so the timer will fire.
+				clock.tick(10); // Time is now 1250
+
+				// expect(runtime.submittedSignals).toHaveLength(2);
+			});
+
+			it("batches signals with different allowed latencies", async () => {
+				runtime.signalsExpected.push([
+					"Pres:DatastoreUpdate",
+					{
+						"sendTimestamp": 1060,
+						"avgLatency": 10,
+						"data": {
+							"system:presence": {
+								"clientToSessionId": {
+									"client2": {
+										"rev": 0,
+										"timestamp": 1000,
+										"value": "sessionId-2",
+									},
+								},
+							},
+							"s:name:testStateWorkspace": {
+								"count": {
+									"sessionId-2": {
+										"rev": 2,
+										"timestamp": 1050,
+										"value": {
+											"num": 34,
+										},
+									},
+								},
+								"note": {
+									"sessionId-2": {
+										"rev": 1,
+										"timestamp": 1020,
+										"value": {
+											"message": "will be queued",
+										},
+									},
+								},
+							},
+						},
+					},
+				]);
+
+				// Configure a state workspace
+				const stateWorkspace = presence.getStates("name:testStateWorkspace", {
+					count: Latest({ num: 0 }, { allowableUpdateLatencyMs: 100 }),
+					note: Latest({ message: "" }, { allowableUpdateLatencyMs: 50 }),
+				}); // will be queued, deadline is set to 1060
+
+				const { count, note } = stateWorkspace.props;
+
+				clock.tick(10); // Time is now 1020
+				note.local = { message: "will be queued" }; // will be queued, deadline remains 1060
+				count.local = { num: 12 }; // will be queued; deadline remains 1060
+
+				clock.tick(30); // Time is now 1050
+				count.local = { num: 34 }; // will be queued; deadline remains 1060
+
+				clock.tick(10); // Time is now 1060
+				note.local = { message: "final message" }; // will be queued; deadline remains 1060
+
+				// SIGNAL #1
+				// At time 1060, the deadline timer will fire and send a single signal with the value
+				// from the last signal (num=34, message="final message").
+
+				// It's necessary to tick the timer beyond the deadline so the timer will fire.
+				clock.tick(30); // Time is now 1080
+
+				// expect(runtime.submittedSignals).toHaveLength(1);
```

Comment:
There should still only be one signal, even if two timers get scheduled. The first scheduled timer fires at time 1060,
which is also when the second message gets queued, so the second gets batched but then the timer sends it immediately.
The second timer should no-op when it fires. Is that worth checking?

Reply:
It would be worth checking that there are no unexpected updates in general.
But here you would see updates. The `SIGNAL #1` comment should be with `clock.tick(10); // Time is now 1060` (which is the deadline).
See that after that clock advance there is `note.local = { message: "final message" }` and that state does not appear in any sent signal. If the final clock tick was bigger it would be sent. The final clock ticks should generally be large numbers to make sure there isn't anything left around.

Reply:
> See that after that clock advance there is note.local = { message: "final message" } and that state does not appear in any sent signal. If the final clock tick was bigger it would be sent. The final clock ticks should generally be large numbers to make sure there isn't anything left around.

facepalm... I completely missed that the expected signal had the wrong message... Maybe I got it mixed up with another signal in all my scrolling. Regardless, I'll clean this up.